pos_accounting/src/pool/operations.rs

```diff
@@ -92,7 +92,7 @@ pub trait PoSAccountingOperations {
 
     fn decommission_pool(&mut self, pool_id: PoolId) -> Result<PoSAccountingUndo, Error>;
 
-    fn increase_pool_balance(
+    fn increase_pool_pledge_amount(
```

Comment:
I'm confused about this. The pledge amount and the pool balance are two different things. When I said add the staker balance to the pool data, I meant as another field. We need to do this distinction. On one hand, if we have no total balance of the pool, how are we gonna determine the reward? By looping over all shares? That's O(N). Bad. If we don't have the owner balance, then we can't know how much to pay him on decomission.

Reply:
Let's have a call about this tomorrow.

Reply:
pool_balance is a separate field no changes here. It is a sum of pledge_amount and all delegation balances. Every time `increase_pool_pledge_amount` is called it changes `pledge_amount` inside pool_data and also updates pool_balance.

So right now getting total pool balance and pledged amount should be O(logN) - at least for in_memory with BTreeMap. At the same time getting total delegations balance is O(N) because we need to sum them up every time.

Reply:
As discussed, we'll have tests that ensure that the balance distribution is correct.

The misunderstanding here is that we increase the individual balances (for delegators), including the pledge.